src/utils.ts

```diff
@@ -81,14 +87,18 @@ export async function generatePDF({
         await new Promise((r) => setTimeout(r, waitForRender));
       }
 
-      // Get the HTML string of the content section.
-      const html = await getHtmlContent(page, contentSelector);
-
       // Make joined content html
       if (excludeURLs && excludeURLs.includes(nextPageURL)) {
         console.log(chalk.green('This URL is excluded.'));
+      } else if (filterKeyword && !(await matchKeyword(page, filterKeyword))) {
+        console.log(
+          chalk.yellowBright(
+            `Page excluded by keyword filter: ${filterKeyword}`,
+          ),
+        );
       } else {
-        contentHTML += html;
+        // Get the HTML string of the content section.
+        contentHTML += await getHtmlContent(page, contentSelector);
```

Comment:
I moved the getHtmlContent here, as it only has to be done when the page is not excluded...

Reply:
Good catch !